app/models/furniture.rb

```diff
@@ -24,6 +24,20 @@ class Furniture < ApplicationRecord
 
   validates :furniture_kind, presence: true
 
+  # Used by child classes that require storage of secrets, like Square API keys
+  # for a Marketplace
+  has_encrypted :secrets, type: :json
+
+  # Forces consistent access, rather than having to work with a nil value
+  after_initialize do
```

Comment:
```suggestion
  # Setting `secrets` to an empty hash after initialization will force consistent access for developers, rather than having to code around a potential nil value
  after_initialize do
```